### PR TITLE
SERVER-76111: the common users(with readWrite role on system db) can modify the system collection's data,it is very dangerous,this can cause serious problems.normal collection perfect

### PR DESCRIPTION
**Pull request by y123456yz**
When we create an account that can read and write the system db(admin,config,local), the account can modify the system namespace(config.transactions,config.chunks.xxx,config.cache.xx, etc).The reason is that we missed some system namespace when we judge the nornal collection.

### CODE REVIEW

**Comment by y123456yz** [comment]
https://jira.mongodb.org/browse/SERVER-76111

**Comment by sgolemon** [comment]
My initial take on this PR is :+1: as it moves us to a more default-closed position regarding these namespaces, however I would be generally suspect of granting `read` or `readWrite` permissions on the admin/local/config databases to common, non-special users.  I'll discuss this with the team and run some extra regression tests, my primary negative concern being breaking existing, valid use cases.

File: src/mongo/db/namespace_string.h [hunk 1]
```diff
@@ -298,7 +299,40 @@ class NamespaceString {
...
+                || coll() == "tenantMigrationRecipients" || coll() == "external_validation_keys"
+                || coll() == "reshardingOperations" || coll() == "localRenameParticipants"
+                || coll() == "vectorClock" || coll() == "collection_critical_sections"
+                || coll() == "image_collection")) {
+            return false;
+        }
+
+        if (isMongos() || serverGlobalParams.clusterRole == ClusterRole::ConfigServer) {
+            if (isConfigDB() 
+                && (coll() == "chunks" || coll() == "databases"
```
Comment (sgolemon) [inline src/mongo/db/namespace_string.h:322]: On the topic of your last push (to address performance), one thing you can do here is add `_sd` to these strings.  This will create `constexpr StringData` literals which among other things encodes the length of the literal into the view, allowing the comparison to run in constant time for strings of differing length rather than linear time.

```suggestion
                && (coll() == "chunks"_sd || coll() == "databases"_sd
```

Note: No need to actually update your PR.  I'll address these minor points as and if this gets merged, just letting you know this is a thing.

Comment (y123456yz) [reply src/mongo/db/namespace_string.h:322]: [sgolemon](https://github.com/sgolemon)
thanks,  👍，I have learned

**Comment by sgolemon** [comment]
Hi, I know this PR has taken.... some time, but I want you to know it's been spent between test runs and discussion with specific teams about the requirements for their individual system collections and the result is that for now at least, these other collections need to remain in the "normal" category, so I must close this PR, but thank you for initiating some great discussions around these collections.
